components/Banner/examples/supplemental/MDCBannerView.m

```diff
@@ -54,124 +69,312 @@ - (instancetype)initWithCoder:(NSCoder *)aDecoder {
+  _trailingButton = trailingButton;
 }
 
 #pragma mark - Property Setter and Getter
 
-- (void)setText:(NSString *)text {
-  self.textLabel.text = text;
+- (BOOL)isImageSet {
+  return self.imageView.image != nil && !self.imageView.hidden;
 }
 
-- (NSString *)text {
-  return self.textLabel.text;
-}
+#pragma mark - KVO
 
-- (void)setImage:(UIImage *)image {
-  if (image) {
-    _image = image;
-    if (!self.iconImageViewContainer) {
-      CGRect iconImageContainerFrame =
-          CGRectMake(0, 0, kIconImageContainerSideLength, kIconImageContainerSideLength);
-      UIView *iconImageViewContainer = [[UIView alloc] initWithFrame:iconImageContainerFrame];
-      UIImageView *iconImageView = [[UIImageView alloc] initWithImage:image];
-      CGRect iconImageFrame = CGRectMake(0, 0, kIconImageSideLength, kIconImageSideLength);
-      iconImageView.frame = iconImageFrame;
-      iconImageView.contentMode = UIViewContentModeScaleAspectFit;
-      iconImageView.clipsToBounds = YES;
-      [iconImageViewContainer addSubview:iconImageView];
-      self.iconImageViewContainer = iconImageViewContainer;
-      self.iconImageView = iconImageView;
-      [self addSubview:iconImageViewContainer];
-    } else {
-      self.iconImageView.image = image;
-    }
-  } else {
-    _image = nil;
-    [self.iconImageViewContainer removeFromSuperview];
-    self.iconImageViewContainer = nil;
-  }
-  [self setNeedsLayout];
-}
+// TODO: KVO trailingButton.hidden and call `setNeedsUpdateConstraints:`.
 
-- (void)setNumberOfButtons:(NSUInteger)numberOfButtons {
-  NSAssert(numberOfButtons <= kNumberOfButtonsLimit,
-           @"%@ class doesn't support more than %lu buttons", NSStringFromClass([self class]),
-           (unsigned long)kNumberOfButtonsLimit);
-  if (numberOfButtons != self.buttons.count) {
-    NSMutableArray *mutableButtons = [self.buttons mutableCopy];
-    if (numberOfButtons > self.buttons.count) {
-      for (NSUInteger index = self.buttons.count; index < numberOfButtons; ++index) {
-        MDCButton *button = [[MDCButton alloc] init];
-        [mutableButtons addObject:button];
-        [self addSubview:button];
-      }
-    } else if (numberOfButtons < self.buttons.count) {
-      for (NSUInteger index = numberOfButtons; index < self.buttons.count; ++index) {
-        MDCButton *lastObject = [mutableButtons lastObject];
-        [lastObject removeFromSuperview];
-        [mutableButtons removeLastObject];
+#pragma mark - UIView overrides
+
+- (CGSize)sizeThatFits:(CGSize)size {
+  MDCBannerViewLayoutStyle style = [self styleForSizeToFit:size];
+  CGFloat frameHeight = 0.0f;
+  switch (style) {
+    case MDCBannerViewLayoutSingleLineStyle: {
+      frameHeight += kTopPaddingSmall + kBottomPadding;
+      [self.textLabel sizeToFit];
+      [self.leadingButton sizeToFit];
+      CGFloat maximumHeight =
+          MAX(CGRectGetHeight(self.textLabel.frame), CGRectGetHeight(self.leadingButton.frame));
+      if (self.isImageSet) {
```

Comment:
lets change this to be [self isImageSet]
. notations is for properties.

Reply:
It is a `readonly` property, so I kept the dot notation for now.